frontend/src/pages/PointsOfInterestView.vue

```diff
@@ -0,0 +1,185 @@
+    import {poiApi} from '@/api.ts';
+    import PointOfInterestCard from '@/components/PointOfInterestCard.vue';
+    import {useI18n} from 'vue-i18n';
+    import {useToast} from 'primevue';
+    import type {FormSubmitEvent} from '@primevue/forms';
+    import type {FormResolverOptions} from '@primevue/forms/form';
+    import type {FormErrors} from '@/types/forms.types.ts';
+    import Select from 'primevue/select';
+
+
+    const {t} = useI18n();
+    const toast = useToast();
+
+    const inEditingMode = ref(0);
+
+    const pois = ref<POIDTO[]>([]);
+    const selectedPOIs = ref<number[]>([])
+    const currentPOI = ref();
+
+    const initialValues = ref({
+        name: '',
+        description: '',
+        cityId: '',
+        price: '',
+        type: ''
+    });
+
+    const types = ref([]);
+    const formKey = ref(0);
+
+    await fetchPOIs();
+    await fetchTypes();
+
+    async function fetchPOIs() {
+        pois.value = await poiApi.getPointsOfInterest();
+    }
+
+    async function fetchTypes() {
+        types.value = await poiApi.getAllPointsOfInterestTypes();
+    }
+
+    function select(poi: POIDTO) {
+        const poiAlreadySelected = selectedPOIs.value.includes(poi.id!);
+        if (poiAlreadySelected) {
+            selectedPOIs.value = selectedPOIs.value.filter((poiId) => poiId !== poi.id);
+            return;
+        }
+        selectedPOIs.value.push(poi.id!);
+    }
+
+    async function deletePOI() {
+        for (const id of selectedPOIs.value) {
+            await poiApi.deletePointOfInterest({id});
+        }
+        selectedPOIs.value = [];
+        await fetchPOIs()
+        inEditingMode.value = 0;
+        initialValues.value = {};
+        formKey.value++;
+    }
+
+    function onCardClick(poi: POIDTO) {
+        inEditingMode.value = 2;
+        initialValues.value = poi;
+        currentPOI.value = poi.id;
+        formKey.value++;
+    }
+
+    function onAddClick(){
+        inEditingMode.value = 1;
+        initialValues.value = {};
+        formKey.value++;
+    }
+
+    function resolver(submitEvent: FormResolverOptions) {
+        const errors: FormErrors = {};
+
+        if (submitEvent.values.name == "") {
+            errors.name = [{message: t('field-is-required')}];
+        }
+        if (submitEvent.values.description == "") {
+            errors.description = [{message: t('field-is-required')}];
+        }
+        if (submitEvent.values.cityId == "") {
+            errors.cityId = [{message: t('field-is-required')}];
+        }
+        if (submitEvent.values.price === "" || submitEvent.values.price === null || submitEvent.values.price === undefined) {
+            errors.price = [{message: t('field-is-required')}];
+        }
+        if (submitEvent.values.type == "") {
+            errors.type = [{message: t('field-is-required')}];
+        }
+
+        return {
+            values: submitEvent.values,
+            errors
+        };
+    }
```

Comment:
Usually, all equalities in JavaScript should be handled using the triple equal operator. `===`.
(oops, I missed that in my PR as well)